app/logic/stake_logic.rb

```diff
@@ -217,7 +217,7 @@ def calculate_apy_for_accounts
       return p unless p.code == 200
 
       # number of epochs in year calculated using the duration of last finished epoch
-      num_of_epochs = number_of_epochs(p.payload[:current_epoch].created_at, p.payload[:previous_epoch].created_at)
+      num_of_epochs = number_of_epochs(p.payload[:network])
```

Comment:
You removed using current_epoch and previous_epoch, but left all the related code. Variables, set_epochs method, assign_epochs method. There's still a lot to do in this story.